querier.cc

```diff
@@ -287,15 +287,15 @@ void querier_cache::insert_querier(
   }
 }
 
-void querier_cache::insert(utils::UUID key, data_querier&& q, tracing::trace_state_ptr trace_state) {
+void querier_cache::insert_data_querier(utils::UUID key, data_querier&& q, tracing::trace_state_ptr trace_state) {
     insert_querier(key, _data_querier_index, _stats, std::move(q), _entry_ttl, std::move(trace_state));
 }
 
-void querier_cache::insert(utils::UUID key, mutation_querier&& q, tracing::trace_state_ptr trace_state) {
+void querier_cache::insert_mutation_querier(utils::UUID key, mutation_querier&& q, tracing::trace_state_ptr trace_state) {
     insert_querier(key, _mutation_querier_index, _stats, std::move(q), _entry_ttl, std::move(trace_state));
 }
```

Comment:
If data_querier and mutation_querier become the same thing, do we need separate caches for them?

(even if we can merge them, I'd postpone it to a later patch since that's not just a mechanical change)

Reply:
One is used for data queries, the other for mutation queries (used during read-repair). A read-repair will (I think) inherit the UUID of the data query it runs on behalf on, so it is beneficial to separate these into separate indexes, so a cross-lookup becomes impossible. Otherwise we have to rely on one of the many checks to pick up that the lookup found the wrong querier.
With that said, technically, if all the checks pass, the querier created originally for the data read can be used for the mutation read, and the other way around, as there is nothing specific about these anymore.

Reply:
If there aren't, then we shouldn't have separate caches. It's just complexity.

Note that there won't be reuse in practice, because reconciliation goes back in time. It's possible that the next page of the data query will reuse the mutation query used to reconcile, but it's not very important or likely.

Of course, shouldn't block this series.

Reply:
Note we have a single cache (the reader concurrency semaphore's inactive reader store), the only difference is which index the querier is added to. The index is an intrusive set.
The querier has tree insert/lookup method pairs for data/mutation/shard queriers. Each has its respective index, but the same underlying store.